group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -5524,5 +5670,2897 @@ private static Record newGroupMetadataRecord(
+            .withGroupId("group-id")
+            .withMemberId(leaderId)
+            .withGenerationId(generationId)
+            .build();
+
+        CompletableFuture<SyncGroupResponseData> syncFuture = new CompletableFuture<>();
+        CoordinatorResult<Void, Record> result = context.sendGenericGroupSync(syncRequest, syncFuture);
+
+        assertEquals(
+            Collections.singletonList(newGroupMetadataRecord(group, MetadataVersion.latest())),
+            result.records()
+        );
+        // Simulate successful write to log.
+        result.appendFuture().complete(null);
+
+        assertTrue(syncFuture.isDone());
+        assertEquals(Errors.NONE.code(), syncFuture.get().errorCode());
+
+        CompletableFuture<JoinGroupResponseData> joinFuture = new CompletableFuture<>();
+        result = context.sendGenericGroupJoin(joinRequest.setMemberId(leaderId), joinFuture);
+
+        assertTrue(result.records().isEmpty());
+        assertTrue(joinFuture.isDone());
+        assertEquals(2, joinFuture.get().generationId());
+        assertEquals(Errors.NONE.code(), joinFuture.get().errorCode());
+    }
+
+    private List<JoinGroupResponseMember> toJoinResponseMembers(GenericGroup group) {
+        List<JoinGroupResponseMember> members = new ArrayList<>();
+        String protocolName = group.protocolName().get();
+        group.allMembers().forEach(member -> {
+            members.add(new JoinGroupResponseMember()
+                .setMemberId(member.memberId())
+                .setGroupInstanceId(member.groupInstanceId().orElse(""))
+                .setMetadata(member.metadata(protocolName)));
+        });
+
+        return members;
+    }
+
+    private void checkJoinGroupResponse(
+        JoinGroupResponseData expectedResponse,
+        JoinGroupResponseData actualResponse,
+        GenericGroup group,
+        GenericGroupState expectedState,
+        Set<String> expectedGroupInstanceIds
+    ) {
+        assertEquals(expectedResponse, actualResponse);
+        assertTrue(group.isInState(expectedState));
+
+        Set<String> groupInstanceIds = actualResponse.members()
+            .stream()
+            .map(JoinGroupResponseData.JoinGroupResponseMember::groupInstanceId)
+            .collect(Collectors.toSet());
+
+        assertEquals(expectedGroupInstanceIds, groupInstanceIds);
+    }
+
+    private void verifySessionExpiration(
+        GroupMetadataManagerTestContext context,
+        GenericGroup group,
+        int timeoutMs
+    ) {
+        Set<String> expectedHeartbeatKeys = group.allMembers().stream()
+            .map(member -> genericGroupHeartbeatKey(group.groupId(), member.memberId())).collect(Collectors.toSet());
+
+        // Member should be removed as session expires.
+        List<ExpiredTimeout<Void, Record>> timeouts = context.sleep(timeoutMs);
+        List<Record> expectedRecords = Collections.singletonList(newGroupMetadataRecord(
+            "group-id",
```

Comment:
The hardcoded value is suspicious here. Is it correct?

Reply:
which one? the group is now in empty state because the one member got kicked out.

Reply:
I meant `group-id`. Is it always the same group id?

Reply:
for all generic group api tests i've stuck to "group-id". this group is created via
`GenericGroup group = context.createGenericGroup("group-id");` so the hardcoded value is correct

Reply:
Would it be better to at least get the name from `group`?